resilience4j-spring/src/test/java/io/github/resilience4j/bulkhead/configure/BulkheadRecoveryTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 lespinsideg
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.bulkhead.configure;
+
+import io.github.resilience4j.TestApplication;
+import io.github.resilience4j.TestDummyService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = TestApplication.class)
+public class BulkheadRecoveryTest {
+    @Autowired
+    @Qualifier("bulkheadDummyService")
+    TestDummyService testDummyService;
+
+    @Test
+    public void testRecovery() {
+        assertThat(testDummyService.sync()).isEqualTo("recovered");
+    }
+
+    @Test
+    public void testAsyncRecovery() throws Exception {
+        assertThat(testDummyService.async().toCompletableFuture().get(5, TimeUnit.SECONDS)).isEqualTo("recovered");
+    }
+
+    @Test
+    public void testMonoRecovery() {
+        assertThat(testDummyService.mono("test").block()).isEqualTo("test");
+    }
+
+    @Test
+    public void testFluxRecovery() {
+        assertThat(testDummyService.flux().blockFirst()).isEqualTo("recovered");
+    }
+
+    @Test
+    public void testObservableRecovery() {
+        assertThat(testDummyService.observable().blockingFirst()).isEqualTo("recovered");
+    }
+
+    @Test
+    public void testSingleRecovery() {
+        assertThat(testDummyService.single().blockingGet()).isEqualTo("recovered");
+    }
+
+    @Test
+    public void testCompletableRecovery() {
```

Comment:
what is the recovery handling here for Compeltable ? why it should be null ?

Reply:
Since `Completable.blockingGet()` returns `Throwable` or `null` depends on the terminated result, I wanted to assert that the exception thrown in the method is recovered.